.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+on:
+  workflow_dispatch:
+jobs:
+  publish-release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Ensure valid tag reference
+        if: ${{ github.ref_type != 'tag' }} || ${{ !startsWith(github.ref_name, 'v') }}
```

Comment:
This is going to make testing it from a PR tough so maybe I should hold off on this check.

Reply:
Maybe adding `--dry-run` to the `poetry publish` command would help with testing https://python-poetry.org/docs/cli/#options-8

Reply:
Moved the `setFailure` to `notice` (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions)

added `--dry-run` to `poetry publish`